src/Illuminate/Concurrency/ProcessDriver.php

```diff
@@ -47,9 +48,31 @@ public function run(Closure|array $tasks): array
             $result = json_decode($result->output(), true);
 
             if (! $result['successful']) {
-                throw new $result['exception'](
-                    $result['message']
-                );
+                $exceptionClass = $result['exception'];
+                $reflection = new ReflectionClass($exceptionClass);
+
+                $args = $result['params'] ?? [];
+
+                if (empty($args)) {
+                    throw new $result['exception'](
+                        $result['message']
+                    );
+                }
+
+                $constructor = $reflection->getConstructor();
+                if ($constructor) {
+                    $constructorParams = $constructor->getParameters();
+                    $finalArgs = [];
+
+                    foreach ($constructorParams as $param) {
+                        $paramName = $param->getName();
+                        $finalArgs[] = $args[$paramName] ?? ($param->isOptional() ? $param->getDefaultValue() : null);
+                    }
+                } else {
+                    $finalArgs = [];
+                }
+
+                throw $reflection->newInstanceArgs($finalArgs);
```

Comment:
This doesn't work when you use a normal Exception such as :

```php 
        [$result1, $result2] = Concurrency::run([
            fn () => throw new \Exception('Example'),
            fn () => 'Task 2 completed',
        ]);
```
_Perhaps_ you could cut out all the reflection stuff and just do something like the below, as we already have the params?
```suggestion
                throw new $result['exception'](
                    ...! empty(array_filter($result['params']))
                        ?  $result['params']
                        : [$result['message']]
                );
```

Failing test : 

```php
    public function testRunHandlerProcessErrorWithNormalException()
    {
        $this->expectException(\Exception::class);
        $this->expectExceptionMessage('This is a different exception');
        
        Concurrency::run([
            fn () => throw new Exception(
                'This is a different exception',
            ),
        ]);
    }
```

Reply:
Thanks @jackbayliss , Issue resolved and test case added with default Exception class